dot/sync/worker_test.go

```diff
@@ -9,53 +9,84 @@ import (
 	"time"
 
 	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/lib/common/variadic"
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/stretchr/testify/require"
-	"go.uber.org/mock/gomock"
 )
 
 func TestWorker(t *testing.T) {
 	peerA := peer.ID("peerA")
-	ctrl := gomock.NewController(t)
-
-	reqMaker := NewMockRequestMaker(ctrl)
-	reqMaker.EXPECT().
-		Do(peerA, nil, gomock.AssignableToTypeOf((*network.BlockResponseMessage)(nil))).
-		DoAndReturn(func(_, _, _ any) any {
-			time.Sleep(2 * time.Second)
+	m := uint32(60)
+	blockReq := &network.BlockRequestMessage{
+		RequestedData: 1,
+		StartingBlock: variadic.Uint32OrHash{},
+		Direction:     3,
+		Max:           &m,
+	}
+	requestCounter := 0
```

Comment:
That's code smell that we don't want, the mock struct gives us a way to ensure that a method was called N times, for example, in the following snippet if the `Do` method is not called 2 times the test will fail, so another point to avoid using a handmade `fakeReqMaker` struct

```
reqMaker.EXPECT().
		Do(....).
		DoAndReturn(...).
                Times(2)
```

Reply:
I will revert it